### PR TITLE
Next js - Complete resources 

### PR DESCRIPTION
**Pull request by chaanakyaaM**
# Pull Request

## Description
Please include a summary of the changes and the related issue.  

I have added next js complete resources under programming languages category
Closes # 463

## Type of change
- [x] Adding a new Resource
- [x] Adding a new Category
- [x] Documentation enhancement or fixes
- [ ] Website enhancement or fixes

## Checklist:
- [x] I read carefully [CONTRIBUTING.md](https://github.com/jfmartinz/ResourceHub/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

## Screenshots (if appropriate):

Please add screenshots to help explain your changes.

### CODE REVIEW

**Comment by chaanakyaaM** [comment]
Kindly let me know if there are any changes to be made.

File: Programming Languages/Next JS/readme.md [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+
+### Tutorials
+
+> These tutorials helps you to learn next js more clearly and easily.
+
+### Introduction
+
+<table width="100%">
+      <tr>
+        <th>Resource</th>
```
Comment (jfmartinz) [inline Programming Languages/Next JS/readme.md:60]:
```suggestion
        <th>Resource Name</th>
```

File: Programming Languages/Next JS/readme.md [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+  - [YouTube Playlists](#youtube-playlists)
+- [Conclusion](#conclusion)
+
+### Roadmap
+
+> This roadmap helps you become a skillful React developer.
+
+<table width="100%">
+      <tr>
+        <th>Resource</th>
```
Comment (jfmartinz) [inline Programming Languages/Next JS/readme.md:43]:
```suggestion
        <th>Resource Name</th>
```

File: Programming Languages/Next JS/readme.md [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+        <td><a href="https://nextjs.org/docs">Next.js Documentation</a></td>
+        <td>This link leads to the official Next.js documentation. Here, users can find detailed information about different versions of Next.js, including release notes, upgrade guides, and documentation for legacy versions. This is particularly useful for developers maintaining projects that use older versions of Next.js and need to reference specific documentation or features.</td>
+      </tr>
+</table>
+
+#### Installation
+
+<table width="100%">
+      <tr>
+        <th>Resource</th>
```
Comment (jfmartinz) [inline Programming Languages/Next JS/readme.md:77]:
```suggestion
        <th>Resource Name</th>
```

File: Programming Languages/Next JS/readme.md [hunk 4]
```diff
@@ -0,0 +1,515 @@
...
+        <td><a href="https://nextjs.org/learn/basics/create-nextjs-app/setup">Create a New Next.js Project</a></td>
+        <td>This page on the Next.js official site provides guidance on starting a new Next.js application. It offers instructions on using the create-next-app CLI tool to quickly set up a new project, along with recommendations for structuring the application and configuring essential settings.</td>
+      </tr>
+</table>
+
+#### Basics
+
+<table width="100%">
+      <tr>
+        <th>Resource</th>
```
Comment (jfmartinz) [inline Programming Languages/Next JS/readme.md:98]:
```suggestion
        <th>Resource Name</th>
```

File: Programming Languages/Next JS/readme.md [hunk 5]
```diff
@@ -0,0 +1,515 @@
...
+        <td><a href="https://nextjs.org/docs/basic-features/pages">Creating Pages in Next.js</a></td>
+        <td>This page on the Next.js documentation explains the fundamentals of creating pages. It details the file-based routing system, where each file in the `pages` directory automatically becomes a route in the application. The page covers dynamic routing, nested routes, and how to use the `Link` component for client-side navigation.</td>
+      </tr>
+</table>
+
+#### Fundamentals
+
+<table width="100%">
+      <tr>
+        <th>Resource</th>
```
Comment (jfmartinz) [inline Programming Languages/Next JS/readme.md:119]:
```suggestion
        <th>Resource Name</th>
```

File: Programming Languages/Next JS/readme.md [hunk 6]
```diff
@@ -0,0 +1,515 @@
...
+        <td><a href="https://nextjs.org/docs/advanced-features/automatic-static-optimization">Automatic Static Optimization</a></td>
+        <td>This guide explains Next.js's automatic static optimization feature, which can be leveraged for efficient data fetching and pagination. It covers the benefits and use cases of static optimization for improving page performance and user experience.</td>
+      </tr>
+</table>
+
+#### linking and routing
+
+<table width="100%">
+  <tr>
+    <th>Resource</th>
```
Comment (jfmartinz) [inline Programming Languages/Next JS/readme.md:168]:
```suggestion
    <th>Resource Name</th>
```

File: Programming Languages/Next JS/readme.md [hunk 7]
```diff
@@ -0,0 +1,515 @@
...
+  </tr>
+</table>
+
+### Advanced topics
+
+#### Next.js CSS
+
+<table width="100%">
+  <tr>
+    <th>Resource</th>
```
Comment (jfmartinz) [inline Programming Languages/Next JS/readme.md:199]:
```suggestion
    <th>Resource Name</th>
```

File: Programming Languages/Next JS/readme.md [hunk 8]
```diff
@@ -0,0 +1,515 @@
...
+    <td><a href="https://nextjs.org/docs/advanced-features/using-mdx">Using MDX with Next.js</a></td>
+    <td>This documentation covers how to use MDX (Markdown with JSX) in a Next.js project. It includes instructions on how to style MDX content with custom CSS and integrate it seamlessly into your Next.js application.</td>
+  </tr>
+</table>
+
+#### Next.js Rendering
+
+<table width="100%">
+  <tr>
+    <th>Resource</th>
```
Comment (jfmartinz) [inline Programming Languages/Next JS/readme.md:236]:
```suggestion
    <th>Resource Name</th>
```

File: Programming Languages/Next JS/readme.md [hunk 9]
```diff
@@ -0,0 +1,515 @@
...
+    <td><a href="https://nextjs.org/docs/advanced-features/custom-server">Custom Server in Next.js</a></td>
+    <td>This guide covers how to create a custom server in Next.js. It explains when and why you might need a custom server and provides examples of how to set one up using Node.js and Express.</td>
+  </tr>
+</table>
+
+#### Nextjs Configuration
+
+<table width="100%">
+    <tr>
+        <th>Resource</th>
```
Comment (jfmartinz) [inline Programming Languages/Next JS/readme.md:269]:
```suggestion
        <th>Resource Name</th>
```

File: Programming Languages/Next JS/readme.md [hunk 10]
```diff
@@ -0,0 +1,515 @@
...
+    <td><a href="https://discord.com/invite/nextjs">Discord - Next.js Community</a></td>
+    <td>The Next.js Discord server is a vibrant community where developers can interact, ask questions, and share their projects and experiences related to Next.js.</td>
+  </tr>
+</table>
+
+### Books
+
+<table width="100%">
+  <tr>
+    <th>Resource</th>
```
Comment (jfmartinz) [inline Programming Languages/Next JS/readme.md:388]:
```suggestion
    <th>Resource Name</th>
```

File: Programming Languages/Next JS/readme.md [hunk 11]
```diff
@@ -0,0 +1,515 @@
...
+  </tr>
+</table>
+
+### Courses
+
+#### Free Courses
+
+<table width="100%">
+  <tr>
+    <th>Resource</th>
```
Comment (jfmartinz) [inline Programming Languages/Next JS/readme.md:415]:
```suggestion
    <th>Resource Name</th>
```

File: Programming Languages/Next JS/readme.md [hunk 12]
```diff
@@ -0,0 +1,515 @@
...
+    <td><a href="https://www.amazon.com/Next-js-Stack-Framework-Development-Course-ebook/dp/B0B97NCTXY">Next.js, The Full Stack React JS Framework, Web Development Crash Course</a></td>
+    <td>This book provides a comprehensive introduction to Next.js, focusing on its capabilities as a full-stack React framework. It covers fundamental topics such as server-side rendering, file-based routing,  API routes and more.</td>
+  </tr>
+</table>
+
+#### Paid Courses
+
+<table width="100%">
+  <tr>
+    <th>Resource</th>
```
Comment (jfmartinz) [inline Programming Languages/Next JS/readme.md:440]:
```suggestion
    <th>Resource Name</th>
```

File: Programming Languages/Next JS/readme.md [hunk 13]
```diff
@@ -0,0 +1,515 @@
...
+
+### YouTube
+
+> These Resources by youtube help you understand Next JS even more clearly.
+
+#### YouTube Channels
+
+<table width="100%">
+      <tr>
+        <th>Resource</th>
```
Comment (jfmartinz) [inline Programming Languages/Next JS/readme.md:465]:
```suggestion
        <th>Resource Name</th>
```

File: Programming Languages/Next JS/readme.md [hunk 14]
```diff
@@ -0,0 +1,515 @@
...
+        <td><a href="https://www.youtube.com/@Codevolution/videos">Codevolution</a></td>
+        <td>This YouTube channel "Codevolution" offers a comprehesive collection of content focused on next.js , web development, and more,it provides the valuable resources for anyone interested in learning web development, improving their coding skills, and staying up-to-date with the latest trends and technologies in the field.</td>
+      </tr>
+</table>
+
+#### YouTube Playlists
+
+<table width="100%">
+      <tr>
+        <th>Resource</th>
```
Comment (jfmartinz) [inline Programming Languages/Next JS/readme.md:490]:
```suggestion
        <th>Resource Name</th>
```

**Review by jfmartinz — CHANGES_REQUESTED** [review]

**Comment by chaanakyaaM** [comment]
Done

**Review by jfmartinz — APPROVED** [review]

**Pull request merged**